### ISSUE TITLE
Java error to create template

### ISSUE THREAD

**Issue by taoansf** [ichthyop/ichthyop#21]
Hello,

I am a new Ichthyop user. 
After a successful installation, the program shows this error when I tryed to create a new project.

ichthyop[severe] Failed to create new configuration file ==> java.io.IOException: Failed to create template at D:\TAOAN\ICTHIOP\simulacao_rwmwd\2021_07_28_Usuario_config_mercactor2d.xml ==> java.lang.NullPointerException: Cannot invoke "java.net.URL.openConnection()" because "url" is null | java.io.IOException: Failed to create template at D:\TAOAN\ICTHIOP\simulacao_rwmwd\2021_07_28_Usuario_config_mercactor2d.xml ==> java.lang.NullPointerException: Cannot invoke "java.net.URL.openConnection()" because "url" is null

I tryed install on linux debian and windows 10, but both systems give me the same problem.

I would appreciate any help to solve this issue.

Thank you for your attention.

With best regards,

Taoan

**Comment by ever-giachini** [comment]
I had the same problem with version 3.3.9.
"Solved" it downloading the previous version.

**Comment by taoansf** [comment]
Thank you for your reply ever-giachini

I also tried downgrading, but maven installer is not working ​with previous versions of ichthyop on my system, I dont know why

I will try the installation using conda enviroment. Did you have experience with conda?

**Comment by ever-giachini** [comment]
Here it worked well when reinstaling. Did you use the same folder name?
I never used conda.

**Comment by taoansf** [comment]
I tried to change folder names and permissions, but without sucess

I use conda enviroment, but same problem.

When I downgrade to 3.3.8 this problem

(ichthyop) D:\TAOAN\ICTHIOP\ichthyop-3.3.8>mvn -B package
[INFO] Scanning for projects...
[INFO]
[INFO] -----------------------< org.previmer:ichthyop >------------------------
[INFO] Building Ichthyop 3.3.8
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/com/akathist/maven/plugins/launch4j/launch4j-maven-plugin/1.7.25/launch4j-maven-plugin-1.7.25.pom
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.869 s
[INFO] Finished at: 2021-07-29T08:20:13-03:00
[INFO] ------------------------------------------------------------------------
[ERROR] Plugin com.akathist.maven.plugins.launch4j:launch4j-maven-plugin:1.7.25 or one of its dependencies could not be resolved: Failed to read artifact descriptor for com.akathist.maven.plugins.launch4j:launch4j-maven-plugin:jar:1.7.25: Could not transfer artifact com.akathist.maven.plugins.launch4j:launch4j-maven-plugin:pom:1.7.25 from/to central (https://repo.maven.apache.org/maven2): Transfer failed for https://repo.maven.apache.org/maven2/com/akathist/maven/plugins/launch4j/launch4j-maven-plugin/1.7.25/launch4j-maven-plugin-1.7.25.pom: Remote host terminated the handshake: SSL peer shut down incorrectly -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginResolutionException

**Comment by ever-giachini** [comment]
Ah, I just realised I also reinstalled JDK between the install of the two versions. Maybe it is the cause I did not get the error.

**Comment by taoansf** [comment]
Ok, I will try to downgrade JDK ;)

**Comment by barriern** [comment]
Hi everyone. 
Regarding the latest errors, it is something due to the `pom.xml` file.You need to remove the given lines:
```
            <!-- Create a ".exe" file -->
            <plugin>
                <groupId>com.akathist.maven.plugins.launch4j</groupId>
                <artifactId>launch4j-maven-plugin</artifactId>
                <version>1.7.25</version>
                <executions>
                    <execution>
                        <id>l4j-clui</id>
                        <phase>package</phase>
                        <goals>
                            <goal>launch4j</goal>
                        </goals>
                        <configuration>
                            <headerType>console</headerType>
                            <outfile>${project.build.directory}/${project.build.finalName}.exe</outfile>
                            <jar>${project.build.directory}/${project.build.finalName}.jar</jar>
                            <errTitle>encc</errTitle>
                            <classPath>
                                <mainClass>${mainClass}</mainClass>
                                <addDependencies>true</addDependencies>
                                <preCp>anything</preCp>
                            </classPath>
                            <jre>
                                <minVersion>${maven.compiler.source}.0</minVersion>
                                <opts>
                                    <opt>-Djava.endorsed.dirs=./endorsed</opt>
                                </opts>
                            </jre>
                            <versionInfo>
                                <fileVersion>${maven.compiler.source}.0.0</fileVersion>
                                <txtFileVersion>${maven.compiler.source}</txtFileVersion>
                                <fileDescription>${project.name}</fileDescription>
                                <copyright>CeCILL-B</copyright>
                                <productVersion>${maven.compiler.source}.0.0</productVersion>
                                <txtProductVersion>${maven.compiler.source}</txtProductVersion>
                                <productName>${project.build.finalName}</productName>
                                <originalFilename>${project.build.finalName}.exe</originalFilename>
                                <!-- internalName is mandatory -->
                                <internalName>${project.build.finalName}</internalName>
                            </versionInfo>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
```
This has been between 3.3.8 and 3.3.9.

**Comment by barriern** [comment]
Regarding the template problem, I will take a look and will come back to you.

**Comment by taoansf** [comment]
Thank you barriern! now I can compile the 3.3.8 and it is working as ever-giachini sad

obs: I did not need to downgrade jdk

**Comment by barriern** [comment]
I have tried on my system and it worked. Which version of JDK do you use?

Ichthyop is being developped using OpenJDK 11. Could you try to see if you can make the 3.3.9 work with the same JDK as me?

Thanks

Nicolas

**Comment by taoansf** [comment]
Yes I am using the last openjdk 11 from anaconda, but no sucess with 3.3.9

**Comment by barriern** [comment]
Hi everyone. I just solved the issue in a new release (73db2785c8394e9b2eaa0b7a17c50cf8d6167094).

Could you please tell me if it works? If so I will close the issue

**Comment by ever-giachini** [comment]
It worked well.
Thank you barriern.
